### DISCUSSION TITLE
json file to unorerder_map

### DISCUSSION THREAD

**Discussion by Tomatin** [nlohmann/json#2786] [General]
Hi, I have a json file containing an array of string array
`[
        [
            "mercury",
            "one"
        ],
		[
            "venus",
            "two"
        ],
		[
            "earth",
            "three"
        ]
]`

And I am using get method to assign it to an unordered_map

```
#include <string>
#include <sstream>
#include <iostream>
#include <fstream>
#include <iomanip>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <algorithm>
#include <regex>

#include "json.hpp"

using json = nlohmann::json;

void test_json(void)
{
  json j;

  // Read json 
  try
  {
    std::ifstream in("test.json");
    in >> j;
  }
  catch (const std::exception &e)
  {
  }

  // Convert json object to unordered_map
  std::unordered_map map = j.get<std::unordered_map<std::string, std::string>>();
}

```
But an exception is fired when I try to use the get method

```
void from_json(const BasicJsonType& j, ConstructibleObjectType& obj)
{
    if (JSON_HEDLEY_UNLIKELY(!j.is_object()))
    {
        **JSON_THROW(type_error::create(302, "type must be object, but is " + std::string(j.type_name()), j))**;
    }
```
Thanks in advance !

**Comment by nlohmann** [comment]
The library does not automatically convert from a list of pairs to a C++ associative container. I'm afraid you need to provide a dedicated conversion function for that.
